### ISSUE TITLE
Enhancing Localization and UX for the Query Total Block

### ISSUE THREAD

**Issue by sarthaknagoshe2002** [WordPress/gutenberg#68194]
## Problem Statement  
The current implementation of the Query Total block, while functional, poses challenges for cross-language feasibility and localization. The main issues include:  
1. **String Concatenation**: The use of concatenated strings such as "Displaying," "of," or "result found" is difficult to translate accurately across languages. This approach fails to accommodate grammatical and linguistic differences.  
2. **Pluralization**: Singular and plural forms are not well-supported. Some languages have multiple plural forms, while others use only one, making the current implementation inadequate for proper localization.  
3. **Customizable Labels**: The absence of an intuitive way to allow users to customize labels while maintaining translation accuracy.  
4. **Flexibility in Text Placement**: Prefixes, suffixes, or other linguistic structures vary across languages, requiring more dynamic handling of text positioning.  

## Discussed Solutions

1. **Dynamic Placeholder System** (Proposed in the PR #67994):  
   - Introduce a single text box with support for placeholders (e.g., `{range}`, `{total}`, etc.).  
   - Allow users to define text like `Displaying {range} of {total} results`, where `{range}` and `{total}` are dynamically replaced at runtime.  

2. **Reordering Text Segments**:  
   - Instead of relying on multiple static text boxes, implement a draggable interface where placeholders can be rearranged to fit language-specific structures (e.g., `{prefix} {range} {suffix}`).  

## References  
- [Original PR](https://github.com/WordPress/gutenberg/pull/67994) for context.
- https://github.com/WordPress/gutenberg/pull/67994#issuecomment-2553696397
- https://github.com/WordPress/gutenberg/pull/67994#issuecomment-2553779265
- https://github.com/WordPress/gutenberg/pull/67994#issuecomment-2555556382

This issue aims to improve the Query Total block while making the text editable & ensuring cross language support.

CC:- @swissspidy @fabiankaegy

**Comment by karmatosed** [comment]
This needs discussion not design quite yet I think so let's swap the labels for that and see about getting feedback first on approach. I'm cautious to have solutions until we know what doing.

**Comment by jasmussen** [comment]
Taking a stab, it seem easy to consider these cases, but let me know if there are more:

* 1 result found
* 12 results found
* Displaying 1-10 of 12
* Displaying 1-2 of 2

There's good discussion in #67994 that I want to acknowledge. But it's unclear to me that free-text editing is a good idea, as that is hard to reconcile with plurals (result vs. results). 

In that vein, it feels what design or design feedback this might need is very much dependant on what technically is possible, both insofar as this block has shipped to production, but also what's feasible as far as availabl implementation capacity.

But one direction this could go, is a settings panel in the inspector that is similar to the Data Format picker in WordPress:

<img width="485" height="227" alt="Image" src="https://github.com/user-attachments/assets/85f18af6-5324-430f-b02d-ebbff1cf6da7" />

It could start with just two options, and no _custom_ option at all. If set to "Total results", perhaps:

* 12 results found
* 12

If set to "Range display", perhaps:

* Displaying 1-10 of 12
* 1-10/12

The idea, to start, would be to try and provide an agnostic alternative to the default, and if you wanted any prefixes you could use a row block and type something like:

Results: 12, or Pages: 1-10/12.

**Comment by sarthaknagoshe2002** [comment]
I agree that the free text and placeholder approach is highly problematic from an i18n perspective, and the predefined picker concept feel like the right direction.

The original intent behind the editable text was to allow users to display specific content types, changing "12 results found" to "12 posts found" or "12 movies found". We can still achieve this exact flexibility without any user typed text by using a Format and Noun matrix.

Since every registered post type already carries translated singular and plural labels, we can pull these dynamically via the post type object and pass them directly into standard WordPress translation functions.

The controls would feature two simple options. The first option is Format, with choices for "Total results" and "Range display". The second option is Noun, with choices for "Generic label" and "Post type label".

Selecting the post type label would automatically output safely localized strings like "12 movies found" or "Displaying 1 to 10 of 12 movies". If a query targets multiple post types at once, the block would simply fall back to the generic results label. By featuring this, we might patch almost all the need to have the placeholder to be editable.

This approach completely resolves the complex pluralization and concatenation issues while preserving the exact customization users need. I can update my existing PR to implement this structure if this technical direction sounds good to everyone.
